pinot-query-planner/src/main/java/org/apache/pinot/query/type/TypeFactory.java

```diff
@@ -81,14 +81,14 @@ private RelDataType toRelDataType(FieldSpec fieldSpec) {
         return fieldSpec.isSingleValueField() ? createSqlType(SqlTypeName.DECIMAL)
             : createArrayType(createSqlType(SqlTypeName.DECIMAL), -1);
       case JSON:
-        // TODO: support JSON, JSON should be supported using a special RelDataType as it is not a simple String,
-        // nor can it be easily parsed as a STRUCT.
+        return createSqlType(SqlTypeName.VARCHAR);
```

Comment:
let's separate the support of the JSON type into a separate PR. and for this PR let's only contribute the proper error message with unsupported datatype propagation.

Reply:
Make sense.